### ISSUE TITLE
Weird listwise training loss

### ISSUE THREAD

**Issue by TDDFT** [tensorflow/ranking#120]
I am training using `approx_ndcg_loss` as the loss. I see very weird results from tensorboard. I see the gradient decreases in the first few steps and then steadily increases. On the other hand, the loss function for train and eval decreases. Is this expected for `approx_ndcg_loss` loss function?
![image](https://user-images.githubusercontent.com/5835058/65727028-6dd04900-e06b-11e9-9e43-f2b132280284.png)
![image](https://user-images.githubusercontent.com/5835058/65727057-850f3680-e06b-11e9-97e5-f4194d04537b.png)

**Comment by xuanhuiwang** [comment]
@HongleiZhuang, assign this to you. I remember that you've seen something similar.

**Comment by HongleiZhuang** [comment]
Hi Degao,

There are multiple factors that could lead to this behavior, e.g. learning rate, optimizer.  Empirically, when `Adam` is used as optimizer with `approx_ndcg_loss`, it is likely to observe unstable behaviors.  If this is the case, you may try to switch to `AdaGrad` optimizer.  Also, using batch normalization would make the training more stable in many cases.  Can you provide more details about what parameter configurations you are using so we may dig deeper?

Thanks!

**Comment by TDDFT** [comment]
@HongleiZhuang Hi, I was using adagrad and approx_ndcg_loss. I added batch normalization for each layer and it is more stable now.

**Comment by bendersky** [comment]
Thanks for reporting! Closing this for now.
